deluge/core/torrentmanager.py

```diff
@@ -795,6 +796,22 @@ def load_resume_data_file(self):
             return resume_data
 
     def save_resume_data_file(self):
+        """
+        Save resume data to file in a separate thread to avoid blocking main thread.
+
+        If a save file task is already running, this call is queued.
```

Comment:
The reasoning for stacking the calls is that the resume data is dependent on callbacks from libtorrent. In theory, a `_save_resume_data_file()` call may be running when a shutdown is initiated. If all `save_resume_data alerts` triggered by the call to `save_resume_data()` in `torrentmanager.stop()` are finished before the `_save_resume_data_file()` call (already running when shutting down) has finished, the new call to `_save_resume_data_file()` initiated by the shutdown would be ignored.

With that said, the problem with stacking calls in general did cross my mind, so I think the best solution is to only stack calls when `flush_disk_cache` is true, i.e. on shutdown.